Automattic/wp-calypso
# Plugins: Divide plugins main into smaller components

Part of https://github.com/Automattic/wp-calypso/issues/941

/plugins/ main fine `/plugins/main.jsx` have grow too large (almost 900 LoC). This PR starts trimming the code in that component to extract some components from it.

Here, we create a new component `plugin-list-header` to contain the header and the buttons in the top of each plugin list and all the logic associated with it:

![image](https://cloud.githubusercontent.com/assets/1554855/11933499/82e39492-a7fc-11e5-8481-22251ed40da0.png)

![image](https://cloud.githubusercontent.com/assets/1554855/11933505/8dc90d2e-a7fc-11e5-82dd-9bb09b63f994.png)


How to test
========
1. go to http://calypso.localhost:3000/plugins
2. Make sure that the header bar & the buttons get rendered and works in the same way than in https://wpcalypso.wordpress.com/plugins
